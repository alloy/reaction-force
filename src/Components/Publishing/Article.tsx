import React from "react"
import track from "react-tracking"
import Events from "../../Utils/Events"

import { BannerWrapper } from "./Banner/Banner"
import ArticleWithFullScreen from "./Layouts/ArticleWithFullScreen"
import { ClassicLayout } from "./Layouts/ClassicLayout"
import { NewsLayout } from "./Layouts/NewsLayout"
import { SeriesLayout } from "./Layouts/SeriesLayout"
import { VideoLayout } from "./Layouts/VideoLayout"
import { FullScreenProvider } from "./Sections/FullscreenViewer/FullScreenProvider"
import { ArticleData, DisplayData } from "./Typings"

export interface ArticleProps {
  article: ArticleData
  relatedArticles?: any
  relatedArticlesForPanel?: any
  relatedArticlesForCanvas?: any
  renderTime?: number
  seriesArticle?: ArticleData
  isHovered?: boolean
  isLoggedIn?: boolean
  isMobile?: boolean
  infiniteScrollEntrySlug?: string
  isSuper?: boolean
  isTruncated?: boolean
  emailSignupUrl?: string
  headerHeight?: string
  marginTop?: string | null
  display?: DisplayData
  showTooltips?: boolean
  slideIndex?: number
  tracking?: any
  closeViewer?: () => void
  viewerIsOpen?: boolean
  onOpenAuthModal?: (type: "register" | "login", config: object) => void
  onExpand?: () => void
}

@track(
  (props: ArticleProps) => {
    return {
      page: "Article",
      entity_type: "article",
      entity_id: props.article.id,
    }
  },
  {
    dispatch: data => Events.postEvent(data),
  }
)
export class Article extends React.Component<ArticleProps> {
  getArticleLayout = () => {
    const { article } = this.props

    switch (article.layout) {
      case "classic": {
        return <ClassicLayout {...this.props} />
      }
      case "series": {
        return <SeriesLayout {...this.props} />
      }
      case "video": {
        return <VideoLayout {...this.props} />
      }
      case "news": {
        return <NewsLayout {...this.props} />
      }
      default: {
        return <ArticleWithFullScreen {...this.props} />
      }
    }
  }

  shouldRenderSignUpCta = () => {
    const { article, isLoggedIn, isTruncated, isMobile } = this.props

    return (
      isMobile && article.layout !== "series" && !isLoggedIn && !isTruncated
    )
  }

  render() {
    return (
      <FullScreenProvider>
        {this.getArticleLayout()}
        {this.shouldRenderSignUpCta() && (
          <BannerWrapper article={this.props.article} />
        )}
      </FullScreenProvider>
    )
  }
}
