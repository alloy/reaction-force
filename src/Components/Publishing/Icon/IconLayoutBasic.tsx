import React, { Component } from "react"

export class IconLayoutBasic extends Component<any, null> {
  render() {
    return (
      <svg
        className="layout-basic"
        width="45px"
        height="30px"
        viewBox="0 0 45 30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
          <g className="layout-basic-group" fill={this.props.fill ? this.props.fill : "#000"}>
            <polyline points="6 25 40 25 40 23 6 23" />
            <polyline points="10 30 36 30 36 28 10 28" />
            <rect x="0" y="0" width="45" height="20" />
          </g>
        </g>
      </svg>
    )
  }
}
