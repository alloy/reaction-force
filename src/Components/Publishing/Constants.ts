import cheerio from "cheerio"
import { compact, last, uniq } from "lodash"
import moment from "moment"
import momentTz from "moment-timezone"
import url from "url"
import { ArticleData, DateFormat } from "../Publishing/Typings"

const APP_URL = process.env.APP_URL || "https://www.artsy.net"

/**
 * Matches for Email / Instant Articles
 */
export const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

/**
 * The quality to request from image CDN
 */
export const GLOBAL_IMAGE_QUALITY = 80

/**
 * TODO: Eventually remove sizeMe
 */
export const SIZE_ME_REFRESH_RATE = 500

/**
 * Relative path to article
 */
export const getArticleHref = slug => `/article/${slug}`

/**
 * Absolute path to article
 */
export const getArticleFullHref = slug => `${APP_URL}/article/${slug}`

/**
 * Get the pre-slug part of URL
 */

export const getPreSlugPath = layout => {
  return ["standard", "feature"].includes(layout) ? "article" : layout
}

/**
 * Relative path to editorial entity
 */

export const getEditorialHref = (layout, slug) => {
  const layoutType = getPreSlugPath(layout)
  return `/${layoutType}/${slug}`
}

/**
 * Absolute path to editorial entity
 */
export const getFullEditorialHref = (layout, slug) => {
  const layoutType = getPreSlugPath(layout)
  return `${APP_URL}/${layoutType}/${slug}`
}

/**
 * Absolute path to artsy entity
 */
export const getFullArtsyHref = slug => `${APP_URL}/${slug}`

/**
 * ByLine helpers
 * TODO: Move this into some kind of utils folder
 */
export const getAuthorByline = (authors, isEditoral = true) => {
  const authorCount = Number(authors && authors.length)

  if (authorCount === 1) {
    return authors[0].name || ""
  } else if (authorCount > 1) {
    const names = authors.reduce((prev, curr, i) => {
      let delim
      const len = authors.length
      if (i === len - 1) {
        delim = " and "
      } else if (i === 0) {
        delim = ""
      } else {
        delim = ", "
      }
      return prev + delim + curr.name
    }, "")
    return names

    // No Author
  } else if (isEditoral) {
    return "Artsy Editors"
  }
}

export const getDate = (date, format: DateFormat = "default") => {
  const today = moment()
  const isToday = today.isSame(moment(date), "day")
  const isThisYear =
    moment(date).format("YYYY") === moment(today).format("YYYY")

  switch (format) {
    case "monthDay":
      return moment(date).format("MMM D")
    case "monthYear":
      return momentTz(date)
        .tz("America/New_York")
        .format("MMMM YYYY")
    case "condensed":
      return momentTz(date)
        .tz("America/New_York")
        .format("MMM D, YYYY")
    case "verbose":
      const day = isToday
        ? "Today"
        : momentTz(date)
            .tz("America/New_York")
            .format("MMM D, YYYY")
      const time = momentTz(date)
        .tz("America/New_York")
        .format("h:mm a")
      return `${day} at ${time}`
    case "news":
      return isToday
        ? "Today"
        : isThisYear
          ? momentTz(date)
              .tz("America/New_York")
              .format("MMM D")
          : momentTz(date)
              .tz("America/New_York")
              .format("MMM D, YYYY")
    default:
      return momentTz(date)
        .tz("America/New_York")
        .format("MMM D, YYYY h:mm a")
  }
}

export const getCurrentUnixTimestamp = () => moment().unix()

export const getMediaDate = article => {
  const { published_at, scheduled_publish_at, media } = article
  const { release_date } = media

  if (release_date) {
    return release_date
  } else {
    return published_at || scheduled_publish_at
  }
}

export const formatTime = time => {
  let minutes = Math.floor(time / 60) % 60
  let seconds = Math.floor(time % 60)
  minutes = minutes <= 0 ? 0 : minutes
  seconds = seconds <= 0 ? 0 : seconds

  const minutesStr = minutes < 10 ? "0" + minutes : minutes
  const secondsStr = seconds < 10 ? "0" + seconds : seconds
  return minutesStr + ":" + secondsStr
}

interface SlugsFromArticle {
  artists: string[]
  genes: string[]
}

export const getArtsySlugsFromArticle = (
  article: ArticleData
): SlugsFromArticle => {
  const articleBody = article.sections
    .map(section => {
      if (section.type === "text") {
        return section.body
      }
    })
    .join()

  const artists = uniq(getArtsySlugsFromHTML(articleBody, "artist"))
  const genes = uniq(getArtsySlugsFromHTML(articleBody, "gene"))

  return {
    artists,
    genes,
  }
}

export const getArtsySlugsFromHTML = (
  html: string,
  model: string
): string[] => {
  const $ = cheerio.load(html)

  const slugs = compact($("a")).map(a => {
    const href = $(a).attr("href")
    if (href) {
      if (href.match(`artsy.net/${model}`)) {
        return last(url.parse(href).pathname.split("/"))
      } else {
        return null
      }
    } else {
      return null
    }
  })
  return compact(slugs)
}
