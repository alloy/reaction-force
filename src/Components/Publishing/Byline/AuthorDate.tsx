import React from "react"
import styled, { StyledFunction } from "styled-components"
import { pMedia } from "../../Helpers"
import { getAuthorByline, getDate } from "../Constants"
import { unica } from "Assets/Fonts"
import { ArticleLayout, BylineLayout, DateFormat } from "../Typings"

interface AuthorProps {
  authors?: any
  layout?: BylineLayout
  articleLayout?: ArticleLayout
  color?: string
}

interface DateProps {
  date?: string
  layout?: BylineLayout
  format?: DateFormat
}

export const Author: React.SFC<AuthorProps> = props => {
  const { color, layout } = props
  return (
    <StyledAuthor
      className="author"
      condensed={layout === "condensed"}
      color={color}
      withBullet
    >
      By {getAuthorByline(props.authors)}
    </StyledAuthor>
  )
}

export const Date: React.SFC<DateProps> = props => {
  const condensed = props.layout === "condensed"
  const format = props.format || (condensed ? "condensed" : "default")

  return (
    <Text className="date" condensed={condensed}>
      {getDate(props.date, format)}
    </Text>
  )
}

Author.defaultProps = {
  color: "black",
}

export interface TextProps {
  condensed?: boolean
  withBullet?: boolean
}

const div: StyledFunction<TextProps & React.HTMLProps<HTMLInputElement>> =
  styled.div

const Text = div`
  ${props =>
    props.condensed ? unica("s14", "medium") : unica("s16", "medium")}
  margin: 10px 30px 0 0;
  &.date {
    white-space: nowrap;
  }
  ${props => pMedia.sm`
    ${props.condensed ? unica("s12", "medium") : unica("s14", "medium")}
    ${unica("s14", "medium")}
    margin: 10px 20px 0 0;
  `}
`
const adjustForCondensed = condensed => {
  return condensed
    ? `
  &:before {
    min-width: 8px;
    min-height: 8px;
    margin: 0 5px 1px 0;
  }`
    : ""
}

const bullet = color => {
  return `
    &:before {
      content: "";
      display: inline-block;
      min-width: 10px;
      min-height: 10px;
      border-radius: 50%;
      margin: 6px 10px 1px 0;
      background-color: ${color};
    }
  `
}

export const StyledAuthor = Text.extend`
  ${props => (props.withBullet ? bullet(props.color) : "")} ${props =>
  adjustForCondensed(props.condensed)} ${pMedia.sm`
    &:before {
      min-width: 8px;
      min-height: 8px;
    }
  `};
`
