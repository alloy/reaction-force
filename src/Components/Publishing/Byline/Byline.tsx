import React from "react"
import styled, { StyledFunction } from "styled-components"
import { getArticleFullHref } from "../Constants"
import { Author, Date } from "./AuthorDate"
import { Share } from "./Share"

interface BylineProps {
  article: any
  date?: string
  layout?: string
  color?: string
}

interface BylineContainerProps {
  color: string
}

export const Byline: React.SFC<BylineProps> = props => {
  const { article, color, date } = props
  const { contributing_authors, published_at } = article
  const layout = props.layout || article.layout
  const title = article.social_title || article.thumbnail_title
  const url = getArticleFullHref(article.slug)
  const textColor = layout === "fullscreen" ? "white" : color

  // TODO: Replace `authors` with `contributing_authors` when ready. Also in
  // <BasicHeader />
  return (
    <BylineContainer className="Byline" color={textColor}>
      <Author
        authors={contributing_authors}
        color={textColor}
        layout={layout}
      />

      <Date date={date ? date : published_at} layout={layout} />

      {layout !== "condensed" && (
        <Share url={url} title={title} color={textColor} />
      )}
    </BylineContainer>
  )
}

Byline.defaultProps = {
  color: "black",
}

const Div: StyledFunction<
  BylineContainerProps & React.HTMLProps<HTMLDivElement>
> =
  styled.div

const BylineContainer = Div`
  display: flex;
  flex-wrap: wrap;
  display: flex;
  align-items: flex-end;
  color: ${props => props.color};
`
