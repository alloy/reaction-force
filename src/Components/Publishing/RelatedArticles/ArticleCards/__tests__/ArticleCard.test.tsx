import { Date } from "Components/Publishing/Byline/Date"
import { getDate } from "Components/Publishing/Constants"
import {
  SeriesArticle,
  StandardArticle,
  VideoArticle,
} from "Components/Publishing/Fixtures/Articles"
import { EditableChild } from "Components/Publishing/Fixtures/Helpers"
import { IconVideoPlay } from "Components/Publishing/Icon/IconVideoPlay"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { ArticleCard, ArticleCardContainer } from "../ArticleCard"

describe("ArticleCard", () => {
  let standardArticle
  let videoArticle

  beforeEach(() => {
    videoArticle = VideoArticle
    standardArticle = StandardArticle
  })

  it("renders an article properly", () => {
    const component = renderer
      .create(<ArticleCard article={standardArticle} series={SeriesArticle} />)
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders an article with unpublished media properly", () => {
    const component = renderer
      .create(<ArticleCard article={videoArticle} series={SeriesArticle} />)
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders an article with children properly", () => {
    const component = renderer
      .create(
        <ArticleCard
          article={videoArticle}
          series={SeriesArticle}
          editDate={EditableChild("date")}
          editDescription={EditableChild("description")}
          editImage={EditableChild("image")}
          editTitle={EditableChild("title")}
        />
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it("Renders media duration and play icon if article has media and is published", () => {
    videoArticle.media.published = true
    const component = mount(
      <ArticleCard article={videoArticle} series={SeriesArticle} />
    )

    expect(component.find(IconVideoPlay).length).toBe(1)
    expect(component.text()).toMatch("02:28")
  })

  it("Renders coming soon and available date if article has media and is unpublished", () => {
    videoArticle.media.published = false
    const component = mount(
      <ArticleCard article={videoArticle} series={SeriesArticle} />
    )

    expect(component.find(IconVideoPlay).length).toBe(0)
    expect(component.text()).not.toMatch("03:12")
    expect(component.text()).toMatch("Coming Soon")
    expect(component.text()).toMatch("Available")
  })

  it("Renders publish date if layout does not have media", () => {
    const component = mount(
      <ArticleCard article={StandardArticle} series={SeriesArticle} />
    )

    expect(component.find(IconVideoPlay).length).toBe(0)
    expect(component.text()).not.toMatch("Coming Soon")
    expect(component.text()).toMatch("May 19, 2017")
  })

  it("Does not render byline if article has media", () => {
    const component = mount(
      <ArticleCard article={videoArticle} series={SeriesArticle} />
    )

    expect(component.find(".author").length).toBe(0)
  })

  it("Renders the media release_date if present", () => {
    const component = mount(
      <ArticleCard article={videoArticle} series={SeriesArticle} />
    )
    const renderedDate = component
      .find(Date)
      .at(0)
      .text()
    const formattedDate = getDate(videoArticle.media.release_date, "monthYear")

    expect(renderedDate).toBe(formattedDate)
  })

  it("Renders the published_at date if no release_date and media is present", () => {
    delete videoArticle.media.release_date
    const component = mount(
      <ArticleCard article={videoArticle} series={SeriesArticle} />
    )
    const renderedDate = component
      .find(Date)
      .at(0)
      .text()
    const formattedDate = getDate(videoArticle.published_at, "monthYear")

    expect(renderedDate).toBe(formattedDate)
  })

  it("Renders editable fields if present", () => {
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
        editDate={EditableChild("date")}
        editDescription={EditableChild("description")}
        editImage={EditableChild("image")}
        editTitle={EditableChild("title")}
      />
    )
    expect(component.text()).toMatch("Child date")
    expect(component.text()).toMatch("Child description")
    expect(component.text()).toMatch("Child image")
    expect(component.text()).toMatch("Child title")
  })

  describe("Analytics", () => {
    it("tracks article card click", () => {
      const trackEvent = jest.fn()
      const component = mount(
        <ArticleCard
          article={videoArticle}
          series={SeriesArticle}
          tracking={
            {
              trackEvent,
            } as any
          }
        />
      )
      component
        .find(ArticleCardContainer)
        .first()
        .simulate("click")
      expect(trackEvent).toBeCalledWith({
        action: "Click",
        label: "Related article card",
      })
    })
  })
})
