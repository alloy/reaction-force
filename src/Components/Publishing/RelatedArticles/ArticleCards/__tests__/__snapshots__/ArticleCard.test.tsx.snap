// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders an article properly 1`] = `
.c3 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c8 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c7 {
  margin: 10px 20px 0 0;
}

.c7::before {
  content: "";
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 0 5px 1px 0;
  background-color: black;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: black;
}

.c12 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c10 {
  position: relative;
  width: 50%;
  height: inherit;
  margin-left: 30px;
}

.c0 {
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c11 {
  opacity: 1;
}

.c0:hover .c11 {
  opacity: 0.7;
}

.c0 .c9 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c4 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  margin-bottom: 10px;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
}

@media (max-width:900px) {
  .c10 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c5 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
    margin-bottom: 20px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="new-yorks-next-art-district"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2 c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        The Future of Art
      </div>
      <div
        className="c4"
      >
        New York's Next Art District
      </div>
      <div
        className="c5"
      >
        Land exhibitions, make influential contacts, and gain valuable feedback about your work.
      </div>
    </div>
    <div
      className="Byline c6"
      color="black"
    >
      <div
        className="c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        <div
          className="c7"
          color="black"
        >
          Casey Lesser
        </div>
      </div>
      <div
        className="c8"
      >
        <div
          className="c3"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          May 19, 2017
        </div>
      </div>
    </div>
  </div>
  <div
    className="c9 c10"
  >
    <img
      className="c11 c12"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F7lsxxsw0qPAuKl37jEYitw%252Farticle%2Basset%2B1-hig%2Bres%2Bcopy.jpg&width=680&height=450&quality=95"
    />
  </div>
</a>
`;

exports[`ArticleCard renders an article with children properly 1`] = `
.c3 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c11 {
  width: 32px;
  height: 32px;
}

.c8 {
  position: relative;
  width: 50%;
  height: inherit;
  margin-left: 30px;
}

.c0 {
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .ArticleCard__Image-nqsvxw-0 {
  opacity: 1;
}

.c0:hover .ArticleCard__Image-nqsvxw-0 {
  opacity: 0.7;
}

.c0 .c7 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c4 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  margin-bottom: 10px;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c6 span {
  margin-right: 5px;
}

.c10 {
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c9 svg {
  width: 40px;
}

@media (max-width:900px) {
  .c8 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c5 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c9 svg {
    width: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2 c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        The Future of Art
      </div>
      <div
        className="c4"
      >
        <div>
          Child 
          title
        </div>
      </div>
      <div
        className="c5"
      >
        <div>
          Child 
          description
        </div>
      </div>
    </div>
    <div
      className="c6 c3"
      fontFamily={
        Object {
          "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
          "fontWeight": 500,
        }
      }
      fontSize="14px"
    >
      <div>
        Child 
        date
      </div>
    </div>
  </div>
  <div
    className="c7 c8"
  >
    <div>
      Child 
      image
    </div>
    <div
      className="c9 c10"
    >
      <svg
        className="c11"
        version="1.1"
        viewBox="0 0 40 56"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="black"
            points="0 0 0 56 39.3600006 28"
          />
        </g>
      </svg>
      <div
        className="c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        02:28
      </div>
    </div>
  </div>
</a>
`;

exports[`ArticleCard renders an article with unpublished media properly 1`] = `
.c3 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c6 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c13 {
  width: 32px;
  height: 32px;
}

.c10 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c8 {
  position: relative;
  width: 50%;
  height: inherit;
  margin-left: 30px;
}

.c0 {
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c9 {
  opacity: 1;
}

.c0:hover .c9 {
  opacity: 0.7;
}

.c0 .c7 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c4 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  margin-bottom: 10px;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
}

.c12 {
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c11 svg {
  width: 40px;
}

@media (max-width:900px) {
  .c8 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c5 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c11 svg {
    width: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2 c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        The Future of Art
      </div>
      <div
        className="c4"
      >
        New Study Shows the Gender Pay Gap for Artists Is Not So Simple
      </div>
      <div
        className="c5"
      >
        The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
      </div>
    </div>
    <div
      className="c6"
    >
      <div
        className="c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        Aug 28, 2017
      </div>
    </div>
  </div>
  <div
    className="c7 c8"
  >
    <img
      className="c9 c10"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FwHFgQlrTrHav5O6bQRJ0dg%252FUntitled%2BSuspended_30x67x33%2B%25282%2529_sm%2Bcropped.jpg&width=680&height=450&quality=95"
    />
    <div
      className="c11 c12"
    >
      <svg
        className="c13"
        version="1.1"
        viewBox="0 0 40 56"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="black"
            points="0 0 0 56 39.3600006 28"
          />
        </g>
      </svg>
      <div
        className="c3"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        02:28
      </div>
    </div>
  </div>
</a>
`;
